[2025-05-06T21:15:11.159+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_kafka_pipeline_dag.send_results_to_kafka manual__2025-05-06T21:15:09.653817+00:00 [queued]>
[2025-05-06T21:15:11.169+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_kafka_pipeline_dag.send_results_to_kafka manual__2025-05-06T21:15:09.653817+00:00 [queued]>
[2025-05-06T21:15:11.169+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2025-05-06T21:15:11.211+0000] {taskinstance.py:1327} INFO - Executing <Task(BashOperator): send_results_to_kafka> on 2025-05-06 21:15:09.653817+00:00
[2025-05-06T21:15:11.214+0000] {standard_task_runner.py:57} INFO - Started process 201 to run task
[2025-05-06T21:15:11.217+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'spark_kafka_pipeline_dag', 'send_results_to_kafka', 'manual__2025-05-06T21:15:09.653817+00:00', '--job-id', '81', '--raw', '--subdir', 'DAGS_FOLDER/spark_kafka_pipeline_dag.py', '--cfg-path', '/tmp/tmpwnzd6262']
[2025-05-06T21:15:11.219+0000] {standard_task_runner.py:85} INFO - Job 81: Subtask send_results_to_kafka
[2025-05-06T21:15:11.258+0000] {logging_mixin.py:150} WARNING - /home/airflow/.local/lib/python3.7/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-05-06T21:15:11.297+0000] {task_command.py:410} INFO - Running <TaskInstance: spark_kafka_pipeline_dag.send_results_to_kafka manual__2025-05-06T21:15:09.653817+00:00 [running]> on host 76cae37c5c10
[2025-05-06T21:15:11.438+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='spark_kafka_pipeline_dag' AIRFLOW_CTX_TASK_ID='send_results_to_kafka' AIRFLOW_CTX_EXECUTION_DATE='2025-05-06T21:15:09.653817+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-06T21:15:09.653817+00:00'
[2025-05-06T21:15:11.439+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-05-06T21:15:11.440+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'python /opt/airflow/scripts/pipeline.py']
[2025-05-06T21:15:11.446+0000] {subprocess.py:86} INFO - Output:
[2025-05-06T21:15:12.083+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-05-06T21:15:12.083+0000] {subprocess.py:93} INFO -   File "/opt/airflow/scripts/pipeline.py", line 34, in <module>
[2025-05-06T21:15:12.083+0000] {subprocess.py:93} INFO -     "bootstrap_servers": os.getenv("KAFKA_BOOTSTRAP_SERVERS").split(","),
[2025-05-06T21:15:12.083+0000] {subprocess.py:93} INFO - AttributeError: 'NoneType' object has no attribute 'split'
[2025-05-06T21:15:12.131+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-05-06T21:15:12.144+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/bash.py", line 211, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-05-06T21:15:12.148+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_kafka_pipeline_dag, task_id=send_results_to_kafka, execution_date=20250506T211509, start_date=20250506T211511, end_date=20250506T211512
[2025-05-06T21:15:12.182+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 81 for task send_results_to_kafka (Bash command failed. The command returned a non-zero exit code 1.; 201)
[2025-05-06T21:15:12.198+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-05-06T21:15:12.225+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
